### PR TITLE
rcon: add flooding protection

### PR DESCRIPTION
**Pull request by diamante0018**
Original review: https://github.com/XLabsProject/iw4x-client/pull/654
Although please let me know if the code can be improved.
I also took the liberty of cleanup the format of some other files.

A steam file will have a huge diff, that is because I changed the line endings to our favorite CRLF

### CODE REVIEW

File: src/client/component/rcon.cpp [hunk 1]
```diff
@@ -52,8 +55,51 @@ namespace rcon
 			network::send(target, "print", console_buffer);
 		}
 
+		bool rate_limit_check(const game::netadr_t& address, const int time)
+		{
+			const auto hash = std::hash<game::netadr_t>()(address);
```
Comment (momo5502) [inline src/client/component/rcon.cpp:60]: honestly no need to hash, just use the netadr as key of the map. that is supported

**Review by momo5502 — CHANGES_REQUESTED** [review]
looks good, only one minor thing

File: src/client/component/rcon.cpp (hunk 1 above)
Comment (diamante0018) [reply src/client/component/rcon.cpp]: Done

File: src/client/component/rcon.cpp [hunk 2]
```diff
@@ -57,20 +57,19 @@ namespace rcon
 
 		bool rate_limit_check(const game::netadr_t& address, const int time)
 		{
-			const auto hash = std::hash<game::netadr_t>()(address);
-			if (!rate_limit_map.contains(hash))
+			if (!rate_limit_map.contains(address))
 			{
-				rate_limit_map[hash] = 0;
+				rate_limit_map[address] = 0;
 			}
```
Comment (momo5502) [inline src/client/component/rcon.cpp]: don't think that's necessary, accessing the map in line 65 will default construct the entry anyways if it's not contained. the default value for int should be zero, so i think you can yeet ll 60-63

**Review by momo5502 — CHANGES_REQUESTED** [review]

File: src/client/component/rcon.cpp (hunk 2 above)
Comment (diamante0018) [reply src/client/component/rcon.cpp]: I had to test it to believe it but it's true

Comment (diamante0018) [reply src/client/component/rcon.cpp]: Done

Comment (momo5502) [reply src/client/component/rcon.cpp]: at() throws an exception, operator[] default constructs :D

**Review by momo5502 — APPROVED** [review]

**Pull request merged**